src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -376,6 +377,36 @@ def build_archive_modal_payload(self, callback_id: str, metadata: str) -> View:
             metadata=metadata,
         )
 
+    def _update_view(
+        self,
+        slack_client: SlackSdkClient,
+        external_id: str,
+        modal_payload: View,
+        slack_request: SlackActionRequest,
+    ) -> None:
+        try:
+            slack_client.views_update(
+                external_id=external_id,
+                view=modal_payload,
+            )
+        except SlackApiError as e:
+            # If the external_id is not found, Slack we send `not_found` error
+            # https://api.slack.com/methods/views.update
+            if unpack_slack_api_error(e) == MODAL_NOT_FOUND:
+                # The modal was not found, so we need to open a new one
+                self._open_view(slack_client, modal_payload, slack_request)
```

Comment:
is this something we'll always need to account for?

Reply:
yes, this will help if the first message we send from control fails (ratelimited for example), this will give us another shot to still successfully send the message.

Reply:
hmm should we emit a metric from this? is the first message failing indicative of anything